packages/client/hmi-client/src/services/models/simulation-service.ts

```diff
@@ -250,12 +242,46 @@ export async function simulationPollAction(
 	emitFn: (event: 'append-output-port' | 'update-state', ...args: any[]) => void
 ) {
 	const requestList: Promise<Simulation | null>[] = [];
+
 	simulationIds.forEach((id) => {
 		requestList.push(getSimulation(id));
 	});
 	const response = await Promise.all(requestList);
-	if (response.every((simulation) => simulation!.status === ProgressState.COMPLETE)) {
+
+	const completedSimulationIds = response
+		.filter((simulation) => simulation?.status === ProgressState.COMPLETE)
+		.map((simulation) => simulation!.id);
+	const inProgressSimulationIds = response
+		.filter(
+			(simulation) =>
+				simulation?.status === ProgressState.QUEUED || simulation?.status === ProgressState.RUNNING
+		)
+		.map((simulation) => simulation!.id);
+	const unhandledStateSimulationIds = response.filter(
+		(simulation) =>
+			simulation?.status !== ProgressState.QUEUED &&
+			simulation?.status !== ProgressState.RUNNING &&
+			simulation?.status !== ProgressState.COMPLETE
+	);
+
+	// there are unhandled states - we will return an error
+	if (unhandledStateSimulationIds.length > 0) {
 		const newState = deleteSimulationInProgress(node, simulationIds);
```

Comment:
did some cleaning here to make the if conditions hopefully more readable understandable